### ISSUE TITLE
Cleanup of stratified inline code

### ISSUE THREAD

**Issue by shazqadeer** [boogie-org/boogie#200]
Currently, Boogie implements code used as a library for implementing stratified inlining of procedures.  But it also implements the algorithm itself, without all the bells and whistles of Corral.  It would be good to keep the library but eliminate the implementation of the algorithm to reduce maintenance and increase robustness.

**Comment by shazqadeer** [comment]
@akashlal : Are you planning to look into this issue?

**Comment by shazqadeer** [comment]
@akashlal :  We need to make progress on this issue.  I am happy to work on it myself if you don't have the cycles. For starters, could you tell me which flags in CommandLineOptions need to be retained for Corral's benefit?

**Comment by akashlal** [comment]
Can I blame the delay on the pandemic? :) Give me this week to sort this out ...

**Comment by akashlal** [comment]
I need some tips. I want to locally make some changes to Boogie, and then try out those changes with Corral. The trouble is that Corral now consumes Boogie as a nuget package. What's the best way to try out my local changes?

**Comment by akashlal** [comment]
I think I'm done -- on this branch: akashl/remove-si
Still need to test Corral, but unable to pull in the Boogie dlls ...

**Comment by RustanLeino** [comment]
> I need some tips. I want to locally make some changes to Boogie, and then try out those changes with Corral. The trouble is that Corral now consumes Boogie as a nuget package. What's the best way to try out my local changes?

I had a tough time doing this with Dafny. I was able to do it once, but wasn't able to reproduce it after that. If you figure out the right recipe for doing this, I'd love to know.

**Comment by shazqadeer** [comment]
You should be able to create a Boogie package locally as follows:

dotnet build Source/Boogie.sln
dotnet pack Source/Boogie.sln

After that, I suppose you have to tell Corral to grab the generated package rather than pulling from nuget.  That would require some investigation.

**Comment by shazqadeer** [comment]
@akashlal : Could you please put up a PR so I can look at your changes?

**Comment by RustanLeino** [comment]
> After that, I suppose you have to tell Corral to grab the generated package rather than pulling from nuget. That would require some investigation.

Yeah, this is where I had got stuck.

**Comment by shazqadeer** [comment]
I found a pertinent discussion on stackoverflow: https://stackoverflow.com/questions/43400069/add-a-package-with-a-local-package-file-in-dotnet.

Also, @bkragl may have some insights.

**Comment by akashlal** [comment]
Unfortunately, I'm stuck. I've tried a bunch of things, but I am unable to point Corral to a locally created nuget package. I will ask around for help.

**Comment by akashlal** [comment]
Created PR #336. 

I somehow managed to do this. First, I switched version in Boogie from `2.8.26` to `2.8.26-l` and did the `dotnet pack` to build the boogie package. Next, I switched Corral to use boogie version `2.8.26-l`. I had to instruct it to find the package in a local folder by adding a `NuGet.config` file to `corral\source`, like how its done [here](https://github.com/microsoft/coyote-samples/blob/main/NuGet.config#L9).

But given the fact that I tried a lot of things, I have messed something up on my machine, so suddenly `Gitversiontask` has started giving me grief. Probably starting clean and following the steps written above would work, but I'm not sure.
